hbase-asyncfs/src/main/java/org/apache/hadoop/hbase/io/asyncfs/FanOutOneBlockAsyncDFSOutput.java

```diff
@@ -157,6 +159,13 @@ public Callback(CompletableFuture<Long> future, long ackedLength,
         replicas.stream().map(Channel::id).forEachOrdered(unfinishedReplicas::add);
       }
     }
+
+    @RestrictedApi(explanation = "Should only be called in tests", link = "",
+        allowedOnPath = ".*/src/test/.*")
+    Set<ChannelId> getUnfinishedReplicas() {
```

Comment:
So this is for checking whether we have finished the ack? I think this could also be done by adding a ChannelInboundHandler, as the completed or failed method will be called inside event loop?

Reply:
I would fix it